### PR TITLE
Update Github pull request template to have a task for inspecting failing checks

### PR DESCRIPTION
**Pull request by reta**
### Description
Update Github pull request template to have a task for inspecting failing checks. Since we have not got the consensus on https://github.com/opensearch-project/OpenSearch/issues/10371, and this issue does not only persist but gets worst with mostly every new feature integrated, there is an alternative path to tackle the problem making it a part of the process. 

The idea is to at least ask every contributor to inspect the failing checks (we are talking about Gradle checks primarily) and make a reference (by commenting) to:
 - an existing known issue
 - or, create and attach a new issue 

It won't solve the problem of flaky tests going away but at least we may get attention (hopefully engaging more engineers into fixing flaky tests) and we should at least spot a new flaky tests proactively (since right now most people just push or retrigger the checks).

### Related Issues
- Relates to https://github.com/opensearch-project/OpenSearch/issues/10371

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s)
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
I suspect most people ignore these check boxes, and we already [have text](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/gradle-check.yml#L127) that says the same thing. We can highlight that, or maybe find a way to block PRs where the failures haven't been examined?

**Comment by reta** [comment]
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/28462/

Should be fixed by https://github.com/opensearch-project/OpenSearch/pull/10754

**Comment by peternied** [comment]
Thanks @reta I like documenting expectations for contributors; however, this might be easy to forget for them to do and for maintainers to forget to follow up on - I am only human.

Borrowing from the security repo [[link]](https://github.com/opensearch-project/security/blob/main/MAINTAINERS.md#practices), we have a process of publicly documenting practices to inform intentions.  I'm curious what you think of this approach - maybe you'd want to add a first practice to OpenSearch for this scenario?

**Comment by reta** [comment]
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/28485/

https://github.com/opensearch-project/OpenSearch/issues/10749

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -162,3 +163,14 @@ During the PR process, expect that there will be some back-and-forth. Please try
 If we accept the PR, a [maintainer](MAINTAINERS.md) will merge your change and usually take care of backporting it to appropriate branches ourselves.
 
 If we reject the PR, we will close the pull request with a comment explaining why. This decision isn't always final: if you feel we have misunderstood your intended change or otherwise think that we should reconsider then please continue the conversation with a comment on the PR and we'll do our best to address any further points you raise.
+
+## Troubleshooting Failing Builds
```
Comment (reta) [inline CONTRIBUTING.md:167]: @peternied how does it look to you?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ Resolves #[Issue number to be closed when this PR is merged]
   - [ ] All tests pass
 - [ ] New functionality has been documented.
   - [ ] New functionality has javadoc added
+- [ ] Failing checks are inspected and point to the corresponding known issue(s)
```
Comment (peternied) [inline .github/pull_request_template.md:20]: What do you think about adding link to the section in contributing.md?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -162,3 +163,14 @@ During the PR process, expect that there will be some back-and-forth. Please try
...
+
+The OpenSearch testing framework uses a lot of randomization internally to cover as many edge cases and variations as possible. Unfortunately, this posses a challenge since in most cases, every build is unique and may not trigger a particular flow. It leads straight to so called flaky tests - the tests which flip randomly from success to failure without any code changes.
```
Comment (peternied) [inline CONTRIBUTING.md:169]: I read this a little as random != unreliable test, versus how I see it that OpenSearch has significant complexity which makes it hard to discover important bugs.  What do you think about reshaping the tone?

Comment (reta) [reply CONTRIBUTING.md]: Rephrased, thanks @peternied !

**Review by peternied — APPROVED** [review]

**Comment by reta** [comment]
@dblock are you good with that? :-)
